libs/schema-profunctor/README.md

```diff
@@ -203,6 +203,50 @@ Finally, we add a name to the schema using the `named`
 combinator. This does nothing to the JSON encoding-decoding part of
 the schema, and only affects the documentation.
 
+### Bespoke instances (Untagged sum type with overlapping values)
+
+Sometimes we have specific types handled in very specific ways, requiring
+a highly customised instance.
+
+Here's an example of such a type. We can have either Seconds or Unlimited, 
+with a caveat: 0 ~ Unlimited.
+
+``` haskell
+data AOrB
+  = FeatureTTLSeconds Word
+  | FeatureTTLUnlimited
+  deriving stock (Eq, Show)
+
+instance ToSchema FeatureTTL where
+  schema = mkSchema ttlDoc toTTL fromTTL
+    where
+      ttlDoc :: NamedSwaggerDoc
+      ttlDoc = swaggerDoc @Word & S.schema . S.example ?~ "unlimited"
+
+      toTTL :: A.Value -> A.Parser FeatureTTL
+      toTTL v = parseUnlimited v <|> parseSeconds v
+
+      parseUnlimited :: A.Value -> A.Parser FeatureTTL
+      parseUnlimited =
+        A.withText "FeatureTTL" $
+          \t ->
+            if t == "unlimited" || t == "0"
+              then pure FeatureTTLUnlimited
+              else A.parseFail "Expected ''unlimited' or '0'."
+
+      parseSeconds :: A.Value -> A.Parser FeatureTTL
+      parseSeconds = A.withScientific "FeatureTTL" $
+        \s -> case toBoundedInteger s of
+          Just 0 -> pure FeatureTTLUnlimited
+          Just i -> pure . FeatureTTLSeconds $ i
+          Nothing -> A.parseFail "Expected an integer."
+
+      fromTTL :: FeatureTTL -> Maybe A.Value
```

Comment:
`mkSchema` expects this function to be a maybe. I imagine b/c Aeson.decode can fail.

Reply:
What doesn't make sense to me is this: `fromTTL` goes from the *decoded* value to the json serialization (`Value`).  This is the `encode` operation minus the part from `Value` to `ByteString`, which contains the actually curly brackets and everything.  Nothing should be able to fail there.

I think the fact that you're using `decode` is a mistake.  You don't want to *parse* anything, you want to *render*.  Thereforem my suggestion.

(And now that I look at that again, I wonder if the `fromEnum` is even necessary.  What does `encode @Word` do?  But perhaps the fact that I'm asking myself this question is a good reason to keep the `fromEnum`, because it makes the question unnecessary.)